RealmSwift/SwiftUI.swift

```diff
@@ -645,6 +645,285 @@ extension EnvironmentValues {
+/// a usable state. (see Realm.asyncOpen() documentation)
+///
+/// This property wrapper will publish states of the current `Realm.asyncOpen()` process like progress, errors and an opened realm,
+/// which can be used to update the view
+///
+///     struct AsyncOpenView: View {
+///         @AsyncOpen(appId: "app_id", partitionValue: <partition_value>) var asyncOpen
+///
+///         var body: some View {
+///            switch asyncOpen {
+///            case .notOpen:
+///                ProgressView()
+///            case .open(let realm):
+///                ListView()
+///                   .environment(\.realm, realm)
+///            case .error(_):
+///                ErrorView()
+///            case .progress(let progress):
+///                ProgressView(progress)
+///            }
+///         }
+///     }
+///
+/// This opened `realm` can be later injected to the view as an enviroment value which will be used by our property wrappers
+/// to populate the view with data from the opened realm
+///
+///     ListView()
+///        .environment(\.realm, realm)
+///
+@available(iOS 13.0, macOS 11.0, tvOS 13.0, watchOS 6.0, *)
+@propertyWrapper public struct AsyncOpen: DynamicProperty {
+    @Environment(\.realm) var realm
+    @ObservedObject private var storage = ObservableAsyncOpenStorage()
+
+    private func asyncOpenForUser<T: BSON>(_ user: User, partitionValue: T, configuration: Realm.Configuration) {
+        let userConfig = user.configuration(partitionValue: partitionValue, cancelAsyncOpenOnNonFatalErrors: true)
+        let userSyncConfig = userConfig.syncConfiguration
+        var configuration = configuration
+        configuration.syncConfiguration = userSyncConfig
+        storage.asyncOpen(configuration: configuration)
+            .sink { completion in
+                if case .failure(let error) = completion {
+                    self.storage.asyncOpenState = .error(error)
+                }
+                cancel()
+            } receiveValue: { realm in
+                self.storage.realm = realm
+                self.storage.asyncOpenState = .open(realm)
+            }.store(in: &storage.cancellables)
+    }
+
+    /// :nodoc:
+    public var projectedValue: Published<AsyncOpenState>.Publisher {
+        return storage.$asyncOpenState
+    }
+
+    /// :nodoc:
+    public var wrappedValue: AsyncOpenState {
+        storage.asyncOpenState
+    }
+
+    /**
+     This will cancel any notification from the property wrapper states
+     */
+    public func cancel() {
+        storage.cancellables.forEach { $0.cancel() }
+        storage.cancellables = []
+    }
+
+    /**
+     Initialise the property wrapper
+     - parameter appId: The unique identifier of your Realm app.
+     - parameter partitionValue: The `BSON` value the Realm is partitioned on.
+     - parameter baseConfiguration: A Base Configuration use when creating the Realm,
+     if empty the configuraion is setted to the `defaultConfiguration`
```

Comment:
This needs a bit more explanation because we don't just use the supplied configuration directly. From reading this I would assume that I have to set the syncConfiguration on the config that I pass in.